.github/workflows/python-ci.yml

```diff
@@ -22,20 +22,33 @@ jobs:
         sudo apt-get update
         sudo apt-get install libhdf5-serial-dev netcdf-bin libnetcdf-dev
         sudo apt-get install cdo
-        pip install -U pip pipenv
 
-    - name: Install python packages
+    - name: Install pipenv
       run: |
-        pipenv lock --requirements > requirements.txt
-        pipenv lock --requirements --dev >> requirements.txt
-        pip install -r requirements.txt
+        pip install -U pipenv
+    
+    - id: cache-pipenv
+      uses: actions/cache@v2
+      with:
+        path: ~/.local/share/virtualenvs
+        key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
 
+    - name: Install dependencies if changed
+      if: steps.cache-pipenv.outputs.cache-hit != 'true' && ${{ matrix.python-version == '3.8' }}
+      run: |
+        pipenv install --deploy --dev
+    - name: Re-install dependencies if alternative python version
+      if: ${{ matrix.python-version != '3.8' }}
+      run: |
+        mv Pipfile.lock do-not-use
+        pipenv install --python ${{ matrix.python-version }} --dev
+    
     - name: Test with pytest (full)
       if: github.ref == 'refs/heads/master'
       run: |
-        py.test -v
+        pipenv run py.test -m "not online" -v
```

Comment:
Do we need a checkbox in the PR template for running the whole test suite on your workstation?

Reply:
That would be a good idea.